### ISSUE TITLE
Error in VectorSerializer

### ISSUE THREAD

**Issue by zml2008** [SpongePowered/Sponge#235]
@gabizou a gift for you -- this was me wandering around in the world.
```
[20:58:57] [Server thread/ERROR] [FML]: An Entity type net.minecraft.entity.item.EntityMinecartChest has thrown an exception trying to write state. It will not persist. Report this to the mod author
java.lang.NullPointerException
        at org.spongepowered.mod.util.VectorSerializer.toNbt(VectorSerializer.java:35) ~[VectorSerializer.class:1.8-1354-1.1DEV-0+unknown-b0.git-unknown]
        at net.minecraft.entity.item.EntityMinecart.writeToNbt(EntityMinecart.java:169) ~[adx.class:?]
        at net.minecraft.entity.Entity.onWriteToNBT(Entity.java:597) ~[wv.class:?]
        at net.minecraft.entity.Entity.func_70109_d(Entity.java) ~[wv.class:?]
        at net.minecraft.entity.Entity.func_70039_c(Entity.java:1349) ~[wv.class:?]
        at net.minecraft.world.chunk.storage.AnvilChunkLoader.func_75820_a(AnvilChunkLoader.java:357) [bfy.class:?]
        at net.minecraft.world.chunk.storage.AnvilChunkLoader.func_75816_a(AnvilChunkLoader.java:183) [bfy.class:?]
        at net.minecraft.world.gen.ChunkProviderServer.func_73242_b(ChunkProviderServer.java:246) [qs.class:?]
        at net.minecraft.world.gen.ChunkProviderServer.func_73151_a(ChunkProviderServer.java:305) [qs.class:?]
        at net.minecraft.world.WorldServer.func_73044_a(WorldServer.java:938) [qt.class:?]
        at net.minecraft.server.MinecraftServer.func_71267_a(MinecraftServer.java:363) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:620) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:478) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_25]
```

**Comment by gabizou** [comment]
@zml2008 Actually, this is a perfect example of why checkNotNulls are great ;) 

The [`armorneMod`](https://github.com/SpongePowered/Sponge/blob/master/src/main/java/org/spongepowered/mod/mixin/core/entity/vehicle/minecart/MixinEntityMinecart.java#L169) is being serialized, but it's **supposed** to be initialized [here](https://github.com/SpongePowered/Sponge/blob/master/src/main/java/org/spongepowered/mod/mixin/core/entity/vehicle/minecart/MixinEntityMinecart.java#L53). If that is producing null, well, that's actually an issue for @Mumfrey. I'll try to reproduce the issue if possible. Was it when there's a minecart existing in the world and getting serialized to NBT due to chunk unloading?

**Comment by Mumfrey** [comment]
This should hopefully be fixed by SpongePowered/Mixin@df3d7c5c26daed75cb520c01049014ba05d97350 but I'd appreciate some testing since it's a bit of a hack.

**Comment by Zidane** [comment]
I've not hit this issue using latest Mixin changes. Should we hit another issue in the future, another issue will be opened at that time.
